oak_restricted_kernel/src/mm/encrypted_mapper.rs

```diff
@@ -131,6 +132,52 @@ impl<'a> EncryptedPageTable<MappedPageTable<'a, PhysOffset>> {
             inner: unsafe { MappedPageTable::new(pml4, PhysOffset { offset, encryption }) },
         }
     }
+
+    /// Finds a range of unallocated pages of the requested size.
+    ///
+    /// Args:
+    ///   - start: the pages must start at, or after, `start`
+    ///   - count: number of pages to allocate
+    ///
+    /// Returns:
+    /// The range of unallocated pages, if there was a big enough unallocated gap in the virtual
+    /// address space. The range may start at exactly `start`.
+    pub fn find_unallocated_pages<S: PageSize>(
+        &self,
+        mut start: Page<S>,
+        count: usize,
+    ) -> Option<PageRange<S>> {
+        // This is highly inefficient, but it should be called rarely enough that it doesn't matter
```

Comment:
For now it is called once, so I think it is fine for this PR, but in future user-space might call it multiple times (and in some cases possibly many times). We should think about some data structure to search more efficiently, especially if we start supporting multiple processes.